### PR TITLE
Bii3ddrug

### PR DESCRIPTION
**Pull request by Jack-Ow**
Graphql and React.js changes completed and tested on my PC. Also included Uniprot ID

image

### CODE REVIEW

File: projects/gnomad/src/client/ChorusPages/queries/gnomadVariantQuery.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
+export default `
```
Comment (mhebrard) [inline projects/gnomad/src/client/ChorusPages/queries/gnomadVariantQuery.js]: I would name this file `chorusVariantQuery.js` for clarification

File: projects/gnomad/src/client/VariantPage/GnomadVariantPage.js [hunk 2]
```diff
@@ -146,8 +148,13 @@ const GnomadVariantPage = ({ datasetId, variantId }) => (
                   </p>
                   <MNVSummaryList multiNucleotideVariants={variant.multiNucleotideVariants} />
                 </div>
-              )}
+              )}           
+
             </ResponsiveSection>
+            { datasetId === 'chorus_4k' ? (
```
Comment (mhebrard) [inline projects/gnomad/src/client/VariantPage/GnomadVariantPage.js]: I would integrate BII3DdrugTable as another Annotation block

```js
// Annotations //
</Section>

<Section>
  <h2>Drug Annotations</h2>
  <BII3DdrugTable
    datasetId={datasetId}
    variant={variant}
  />
</Section>

<ResponsiveSection>
// Population Frequencies
```

With a test on the dataset within the BII module

File: projects/gnomad/src/client/ChorusPages/BII3DdrugTable.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    padding-left: 30px;
+    line-height: 1.5;
+  }
+`
+
+const NoWrap = styled.span`
+  white-space: nowrap;
+`
+
+export const BII3DdrugTable = ({ variant }) => {
```
Comment (mhebrard) [inline projects/gnomad/src/client/ChorusPages/BII3DdrugTable.js]:
```suggestion
export const BII3DdrugTable = ({ datasetId, variant }) => {

  if (!datasetId.startsWith('chorus')) {
    return <p>Drug annotation not available for this dataset.</p>
  }

// [...]

  return (
    <Table>
// [...]
  )
}
```

File: projects/gnomad-api/src/schema/datasets/chorus_4k/fetchChorus4kVariantDetails.js [hunk 4]
```diff
@@ -78,7 +78,53 @@ const formatFilteringAlleleFrequency = (variantData, fafField) => {
...
+                { term: { 'REF.keyword' :  ref } },
+                { term: { 'ALT.keyword' :  alt } },
+              ],
+            },
+          },
+        },
+      })
+    )
+
+  if (response.hits.hits.length === 0) {
```
[the remark this answers is not in the document]
Comment (Jack-Ow) [reply projects/gnomad-api/src/schema/datasets/chorus_4k/fetchChorus4kVariantDetails.js]: Done.
https://github.com/c-BIG/gnomadjs/blob/2319143e47a698a792fcaa165edf836a95a7aaed/projects/gnomad-api/src/schema/datasets/chorus_4k/fetchChorus4kVariantDetails.js#L107-L109

File: projects/gnomad-api/src/schema/datasets/chorus_4k/fetchChorus4kVariantDetails.js [hunk 5]
```diff
@@ -168,6 +214,11 @@ const fetchChorus4kVariantDetails = async (ctx, variantId) => {
         transcript_id: `ENST${consequence.transcript_id.toString().padStart(11, '0')}`,
       })
     ),
+   UPID: BII3DdrugVariant.UPID,
```
[the remark this answers is not in the document]
Comment (Jack-Ow) [reply projects/gnomad-api/src/schema/datasets/chorus_4k/fetchChorus4kVariantDetails.js]: This doesn't work, fetching null value for all the fields.
`BII3Ddrug: BII3DdrugVariant`

Changed to this instead.
https://github.com/c-BIG/gnomadjs/blob/2319143e47a698a792fcaa165edf836a95a7aaed/projects/gnomad-api/src/schema/datasets/chorus_4k/fetchChorus4kVariantDetails.js#L216-L219

File: projects/gnomad-api/src/schema/datasets/gnomad_r2_1/GnomadVariantDetailsType.js [hunk 6]
```diff
@@ -142,6 +142,11 @@ const GnomadVariantDetailsType = new GraphQLObjectType({
     },
     rsid: { type: GraphQLString },
     sortedTranscriptConsequences: { type: new GraphQLList(TranscriptConsequenceType) },
+    UPID: { type: GraphQLString },
```
[the remark this answers is not in the document]
Comment (Jack-Ow) [reply projects/gnomad-api/src/schema/datasets/gnomad_r2_1/GnomadVariantDetailsType.js]: Done.

https://github.com/c-BIG/gnomadjs/blob/2319143e47a698a792fcaa165edf836a95a7aaed/projects/gnomad-api/src/schema/index.js#L20

File: projects/gnomad/src/client/ChorusPages/queries/gnomadVariantQuery.js (hunk 1 above)
Comment (Jack-Ow) [reply projects/gnomad/src/client/ChorusPages/queries/gnomadVariantQuery.js]: Done.

image

File: projects/gnomad/src/client/ChorusPages/BII3DdrugTable.js (hunk 3 above)
Comment (Jack-Ow) [reply projects/gnomad/src/client/ChorusPages/BII3DdrugTable.js]: Done.

https://github.com/c-BIG/gnomadjs/blob/fd130244be3d779c9c3e176b99cf3199062ac9ad/projects/gnomad/src/client/ChorusPages/BII3DdrugTable.js#L28-L39

File: projects/gnomad/src/client/VariantPage/GnomadVariantPage.js (hunk 2 above)
Comment (Jack-Ow) [reply projects/gnomad/src/client/VariantPage/GnomadVariantPage.js]: Done.

https://github.com/c-BIG/gnomadjs/blob/fd130244be3d779c9c3e176b99cf3199062ac9ad/projects/gnomad/src/client/VariantPage/GnomadVariantPage.js#L154-L160

**Pull request merged**